### ISSUE TITLE
Fix incompatibility with Field Manager

### ISSUE THREAD

**Issue by engram-design** [spicywebau/craft-neo#462]
Refer to issue https://github.com/verbb/field-manager/issues/78

I believe this is because Neo assumes the field layout for a block type will come from POST params (using `assembleLayoutFromPost`)

https://github.com/spicywebau/craft-neo/blob/2af37c2e9bc10e69c01c0b1ffab01cdfeba3cbf0/src/Field.php#L603-L608

However, in Field Manager, when we clone a field, we take the block types from the origin field, and create new block types based on those, for the new field. As such, the field is all prepped and ready to go, however this code overrides our prepped block types. Because `assembleLayoutFromPost` in this case will equate to null, it resets the field layout.

What I might propose is a check if the field layout is already set for the block type?

```
// Set each block type's field layout based on the data from Craft 3.5's field layout designer
foreach ($this->getBlockTypes() as $blockType) {
    // Check if the block type already has a field layout set - no need to set it again
    if (!$blockType->fieldLayout) {
        $fieldLayout = $fieldsService->assembleLayoutFromPost("types.{$class}.blockTypes.{$blockType->id}");
        $fieldLayout->type = $class;
        $blockType->setFieldLayout($fieldLayout);
    }
}
```

Notably, this wasn't an issue on previous versions - I can't tell you what version exactly, but I was on 2.8.14 before updating to the latest (2.9.11) and this was working

**Comment by ttempleton** [comment]
Yep, your proposed fix looks great.  Would you like to open a pull request, or are you happy for us to make that change?

**Comment by engram-design** [comment]
Happy to do a PR - https://github.com/spicywebau/craft-neo/pull/464

**Comment by ttempleton** [comment]
Thanks again -- just released Neo 2.9.12 with that fix.
